### PR TITLE
fix storage upload

### PR DESCRIPTION
**Pull request by elorzafe**
#### Description of changes:
Update upload browser docs
### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [x] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [ ] JS
- [ ] Swift
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [ ] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://docs.amplify.aws/)` 
            HTML: `<a href="https://docs.amplify.aws/">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/pages/[platform]/build-a-backend/storage/upload/index.mdx [hunk 1]
```diff
@@ -231,7 +231,7 @@ Upload a file in the browser:
 ```javascript
 import { uploadData } from 'aws-amplify/storage';
 
-const uploadDataInBrowser = async () => {
+const uploadDataInBrowser = async (event) => {
```
Comment (reesscot) [inline src/pages/[platform]/build-a-backend/storage/upload/index.mdx:234]: Can we please test this code quickly in a browser? the reference to `filename` on line 238 is likely going to be undefined

File: src/pages/[platform]/build-a-backend/storage/upload/index.mdx [hunk 2]
```diff
@@ -244,7 +244,7 @@ const uploadDataInBrowser = async () => {
 
 ```html
 <h2>Upload File</h2>
-<input type="file" onChange="{()" ="" /> uploadDataInBrowser(event)} />
+<input type="file" onChange={uploadDataInBrowser} />
```
Comment (reesscot) [inline src/pages/[platform]/build-a-backend/storage/upload/index.mdx:247]: Is this supposed to be React code or plain HTML?

File: src/pages/[platform]/build-a-backend/storage/upload/index.mdx (hunk 1 above)
Comment (elorzafe) [reply src/pages/[platform]/build-a-backend/storage/upload/index.mdx:234]: fixed

Comment (ashika112) [reply src/pages/[platform]/build-a-backend/storage/upload/index.mdx:234]: @reesscot Its just a sample code right. It is kinda of assumed that customer passes in their file name in there. Are you suggesting we add a comment here or just make it a string?

File: src/pages/[platform]/build-a-backend/storage/upload/index.mdx (hunk 2 above)
Comment (elorzafe) [reply src/pages/[platform]/build-a-backend/storage/upload/index.mdx:247]: This is react code, and is used presented as html, would make more sense to make it pure html?

File: src/pages/[platform]/build-a-backend/storage/upload/index.mdx (hunk 1 above)
Comment (ashika112) [reply src/pages/[platform]/build-a-backend/storage/upload/index.mdx:234]: Note: Looking through the docs, that how we reference parameter input everywhere

File: src/pages/[platform]/build-a-backend/storage/upload/index.mdx (hunk 2 above)
Comment (ashika112) [reply src/pages/[platform]/build-a-backend/storage/upload/index.mdx:247]: OI suggest we remove this all together.

**Review by ashika112 — APPROVED** [review]

**Review by reesscot — APPROVED** [review]
LGTM, thanks for fixing it!

File: src/pages/[platform]/build-a-backend/storage/upload/index.mdx [hunk 3]
```diff
@@ -231,20 +231,16 @@ Upload a file in the browser:
 ```javascript
 import { uploadData } from 'aws-amplify/storage';
 
-const uploadDataInBrowser = async () => {
+const uploadDataInBrowser = async (event) => {
```
Comment (reesscot) [inline src/pages/[platform]/build-a-backend/storage/upload/index.mdx:234]: does this need to be async?

```suggestion
const uploadDataInBrowser = (event) => {
```

File: src/pages/[platform]/build-a-backend/storage/upload/index.mdx (hunk 1 above)
Comment (reesscot) [reply src/pages/[platform]/build-a-backend/storage/upload/index.mdx:234]: @ashika112 I think the difference here is we are providing what looks like an encapsulated function for handling an browser upload, so we shouldn't be using undefined variables.

File: src/pages/[platform]/build-a-backend/storage/upload/index.mdx (hunk 3 above)
Comment (kvramyasri7) [reply src/pages/[platform]/build-a-backend/storage/upload/index.mdx:234]: +1

**Pull request merged**